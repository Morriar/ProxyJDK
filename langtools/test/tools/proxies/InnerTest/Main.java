/*
 * @test
 * @summary inner classes
 * @author Alexandre Terrasa
 *
 * @compile Main.java
 */

package TMPProxyTests.InnerTest;

public class Main {
	public static void main(String[] args) {
		new Main();
	}
	
	public Main() {
		Runnable prxy = proxy Runnable(new RealRunnable("main"));
		prxy.run();
		
		
		Runnable prxy2 = new RedefRunnable(new RealRunnable("with redef"));
		prxy2.run();
		
		RedefRunnable prxy3 = (RedefRunnable)prxy2;
		prxy3.run();
		prxy3.added();
	}
	
	public class RealRunnable implements Runnable {
		public String name;
		
		public RealRunnable(String name) {
			this.name = name;
		}
		
		public void run() {
			System.out.println("run by " + name);
		}
	}
	
	public class RedefRunnable extends proxy Runnable {

		public RedefRunnable(RealRunnable rs) {
			super(rs);
		}
		
		public void run() {
			super.run();
			System.out.println("modified run");
		}
		
		public void added() {
			System.out.println("added");
		}
	}
}
