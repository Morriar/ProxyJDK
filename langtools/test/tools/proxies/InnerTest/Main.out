run by main
run by with redef
modified run
run by with redef
modified run
added